panic: slice bounds out of range [:8] when GenerateFeedID processes URLs whose CRC32 hex is < 8 chars

## Bug

`GenerateFeedID` in `model/feed_id.go:31` panics with `slice bounds out of range [:8]` when a feed URL's CRC32 hash formats to fewer than 8 hex characters.

The root cause is using `%x` (no zero-padding) instead of `%08x` when formatting the CRC32 value. For certain URLs the hash produces a value like `0x0CE2FCF` which `fmt.Sprintf("%x", ...)` renders as `"ce2fcf"` (6 chars) — then the subsequent `[:8]` slice panics.

## Crash log

```
panic: runtime error: slice bounds out of range [:8] with length 7
goroutine 1044 [running]:
github.com/richardwooding/feed-mcp/model.GenerateFeedID({0x7901ff1b5d00, 0x33})
	/home/runner/work/feed-mcp/feed-mcp/model/feed_id.go:31 +0x1e4
github.com/richardwooding/feed-mcp/store.newStoreInternal.func3({0x7901ff1b5d00, 0x33})
	/home/runner/work/feed-mcp/feed-mcp/store/store.go:513 +0x60
```

## Reproduction

Load an OPML file containing any of these URLs (their CRC32 hex is < 8 chars):

| Feed | URL | CRC32 hex |
|------|-----|-----------|
| O Human Star | `http://ohumanstar.com/feed/` | `ce2fcf` (6 chars!) |
| webcomic name | `http://webcomicname.tumblr.com/rss` | `df20d08` |
| Straight, No Chaser | `http://www.straightnochaser.org/feed/` | `f512e83` |
| Silicon Florist | `http://feeds2.feedburner.com/siliconflorist` | `321dd32` |
| Mondo 2000 | `http://www.mondo2000.com/feed/` | `ae7a306` |

The server starts, pulls the Docker image, begins processing the OPML, then crashes after ~45 seconds — before responding to the MCP `initialize` call.

## Fix

In `model/feed_id.go:31`, change:

```go
// before
fmt.Sprintf("%x", crc)[:8]

// after
fmt.Sprintf("%08x", crc)[:8]
```

Zero-padding to 8 characters ensures the slice never goes out of range.

## Environment

- `ghcr.io/richardwooding/feed-mcp:latest` (v2.2.10)
- macOS, Docker Desktop, Claude Desktop MCP config

--- a/model/feed_id.go
+++ b/model/feed_id.go
@@ -28,7 +28,7 @@ func GenerateFeedID(feedURL string) string {
 		if len(slug) > 40 {
 			h := fnv.New32a()
 			_, _ = h.Write([]byte(feedURL)) // FNV hash Write never returns an error
-			hashStr := fmt.Sprintf("%x", h.Sum32())[:8]
+			hashStr := fmt.Sprintf("%08x", h.Sum32())
 			slug = slug[:32] + "-" + hashStr
 		}
 		return slug

--- a/model/feed_id_test.go
+++ b/model/feed_id_test.go
@@ -35,6 +35,14 @@ func TestGenerateFeedID(t *testing.T) {
 			feedURL:  "https://example.com/news & events/feed.xml",
 			expected: "example.com-news-events-feed-xml", // Spaces and & replaced with single dash
 		},
+		{
+			// Regression test for #113: FNV-32a hash with a leading zero nibble
+			// (Sum32 < 0x10000000) would render as < 8 hex chars under %x and
+			// panic when sliced to [:8]. Zero-padding via %08x prevents this.
+			name:     "Long URL with leading-zero hash",
+			feedURL:  "https://very-long-domain-name-example.com/p22/feed.xml",
+			expected: "very-long-domain-name-example.co-001353e5",
+		},
 	}
 
 	for _, tc := range testCases {

